### ISSUE TITLE
SSL Connections

### ISSUE THREAD

**Issue by TheTexasMiner** [Demion/nodevfee#40]
Question

Does your NoDefFee work with SSL Connections to a pool ?

**Comment by Demion** [comment]
Starting from 11.1 devfee use SSL encryption. No solution so far.  
More details here - https://github.com/Demion/nodevfee/issues/34#issuecomment-370114352.

**Comment by DragonKingz** [comment]
im using claymore 11.2 + nodevfee v0.2.6b mining on ethermine, asia1 :4444 (not using ssl),  still working after i put -allpools 1
:) 
![1520527399394](https://user-images.githubusercontent.com/20707033/37163517-94e36ab8-232a-11e8-8685-da295381a410.jpg)

**Comment by TheTexasMiner** [comment]
Claymore is now using the secure ssl :5555 connection for the devfee starting with 11,3.
So, even if you use the non secure connection :4444 Claymore devfee is using the :5555
for all the devfee connections. This means that you cannot avoid the devfee if using 11.3
or above. Smart of them, not good for nodevfee.

**Comment by jsm0** [comment]
Still no problem running 11.3 on nanopool with divert

**Comment by TheTexasMiner** [comment]
Does nanopool have the ssl connection at this time ?

**Comment by newtoopensuse** [comment]
11.3 actually succeeds in thwarting divert. It seems to know divert is working, so it stops processing shares. I observed that devfee had no shares three times in a row when nodevfee or divert was running. I have stopped using them with claymore 11.3 and 11.4.

**Comment by jsm0** [comment]
This is false. I have tried every newer version of claymore, still using divert 3.x on nanopool. Did you actually check your logs to see if any shares were found during that devfee time? On my 13 gpu rig I consistently get shares every hour as well as my calculated being +/- of reported. I can confirm divert works all the way to current 11.5. SSL is the problem others on ethermine face.

**Comment by Nickduino** [comment]
There's too much confusion between nodevfee and divert here.

**Comment by newtoopensuse** [comment]
@jsm. Confirm what you say. See divert discussion.

**Comment by jsm** [comment]
Wha

**Comment by Demion** [comment]
nodevfee and divert do exactly same thing - replace devfee wallet in network pool authorization packet.

**nodevfee** - use DLL injection and WinApi functions splicing hooks.
**divert** - use WinDivert driver which use _Windows Filtering Platform_ to capture and modify network packets.

There are reports dll injection / api hook method result in lower efficiency starting from 10.6 or 11.0 version.

Starting from 11.1 ethermine devfee use SSL encryption. There is no solution so far. 
If you manage to make it work on other pool (nanopool for instance) for both main and devfee then it should be fine.
Only way to tell if it is working is to monitor your actual effective hashrate on your pool stastics for 24 hours (better more).
@newtoopensuse

You only see 2nd worker (default / empty) when devfee finds share it is determined by luck, so it is random (it may not find any for few times in a row).
Only way to determine if it is working by logs and , as I were saying, your actual effective hashrate on pool stastics (less then 24 hours maybe innacurate).
